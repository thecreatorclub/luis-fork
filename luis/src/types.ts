export let no: string;

declare global {
  interface StoryConfig {
    [index: string]: any;
    component?: React.ComponentType;
  }
  function addStory(story: StoryConfig): void;
}
