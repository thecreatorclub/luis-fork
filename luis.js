#!/usr/bin/env node
// check if custom config exists
const fs = require('fs');
const path = require('path');

// parse parameters
let init = false;
let tests = '';
let root = 'src';
let run = 'luis.';
let language = 'ts';
let customSet = false;
let start = true;
let bundler = 'fusebox';

function rtl(source) {
  const templatePath = require.resolve(`./templates/${source}`);
  // console.log('Loading: ' + templatePath);
  return fs.readFileSync(templatePath, { encoding: 'utf-8' });
}

function launch(program) {
  require('child_process').spawnSync(program, {
    stdio: 'inherit',
    shell: true,
    cwd: path.resolve('.')
  });
}

for (let i = 0; i < process.argv.length; i++) {
  let arg = process.argv[i];
  if (arg === 'init') {
    init = true;
  }
  if (arg === 'start') {
    start = true;
  }
  if (arg === '--tests') {
    tests = process.argv[i + 1];
    i++;
  }
  if (arg === '--language') {
    language = process.argv[i + 1];
    i++;
  }
  if (arg === '--bundler') {
    bundler = process.argv[i + 1];
    i++;
  }
  if (arg === '--root') {
    root = process.argv[i + 1];
    customSet = true;
    i++;
  }
  if (arg === '--run') {
    customSet = true;
    run = process.argv[i + 1];
    i++;
  }
}

if (init) {
  if (!customSet) {
    console.log(
      "No root and entry file specified. Estimating 'src' as project root and 'luis.ts' as the entry files."
    );
    console.log(
      "If you want to use a different entry file, please run process as 'node luis init --root /your/root --run path/to/luis.ts"
    );
  }

  let file = path.resolve(path.join(root, run + language));
  let dir = path.dirname(file);

  if (!fs.existsSync(dir)) {
    console.log('Root directory does not exist');
    console.log('Creating: ' + dir);
    fs.mkdirSync(dir, { recursive: true });
  }

  // load test file
  const configFile = tests ? rtl('entry_tests.js') : rtl('entry_no_tests.js');

  let configPath = path.resolve(file);
  if (fs.existsSync(configPath)) {
    console.log('Luis config already exists');
  } else {
    console.log('Creating: ' + file);
    fs.writeFileSync(configPath, configFile, { encoding: 'utf-8' });
  }

  if (tests === 'jest') {
    let testFilePath = path.resolve(`jest.config.js`);
    if (fs.existsSync(testFilePath)) {
      console.log('Jest config file already exists');
    } else {
      let testFile = language === 'js' ? rtl('jest.config.js') : rtl('jest.config.ts');
      testFile = testFile
        .replace(/\$root/g, root)
        .replace('$ignore', `<rootDir>/${root}/summary.${language}`);
      fs.writeFileSync(testFilePath, testFile, { encoding: 'utf-8' });
      console.log('Creating: ' + testFilePath);
    }
  } else if (tests === 'mocha') {
    console.log('Currently not supported');
  } else if (tests === 'jasmine') {
    console.log('Currently not supported');
  } else {
    console.error('Test not supported: ' + tests);
  }

  let bundlerPath, bundlerConfig;
  if (bundler === 'fusebox') {
    bundlerPath = path.resolve('luis.fuse.js');
    bundlerConfig = rtl('luis.fuse.js')
      .replace('$root', root)
      .replace(/\$entry/g, run + language);
    console.log('Creating: ' + bundlerPath);
  } else if (bundler === 'parcel') {
  } else if (bundler === 'webpack') {
  } else if (bundler === 'cra') {
  } else {
    console.error('Bundler not supported: ' + bundler);
    return;
  }
  fs.writeFileSync(bundlerPath, bundlerConfig, { encoding: 'utf-8' });
  return;
}

const fuseConfig = path.resolve('luis.fuse.js');
const parcelConfig = path.resolve('luis.index.html');
const webpackConfig = path.resolve('luis.webpack.js');
const cra = path.resolve('node_modules/react-scripts');

if (fuseConfig) {
  console.log('Found Fuse-Box config. Fusing luis in 3..2..1');
  launch('node luis.fuse.js');
} else if (parcelConfig) {
} else if (webpackConfig) {
} else if (cra) {
} else {
  console.log(
    'Luis could not find a runtime file for any supported bundler. Please run "luis init" and see the documentation for all cli options'
  );
}
