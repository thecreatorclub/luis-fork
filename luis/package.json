{
  "name": "luis",
  "version": "3.2.13",
  "description": "List-of-uis book and testing of uis",
  "main": "./index.js",
  "types": "./index.d.ts",
  "scripts": {
    "ccopy": "cp -r templates ../dist/ && cp package.json ../dist && cp README.md ../dist",
    "copy": "rm -rf ../dist && tsc && cp -r templates ../dist/templates && cp package.json ../dist && cp README.md ../dist",
    "link": "npm run copy && npm link",
    "publish": "npm run copy && npm publish"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tomitrescak/luis.git"
  },
  "keywords": [
    "ui",
    "react"
  ],
  "author": "tomi.trescak@gmail.com",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/tomitrescak/luis/issues"
  },
  "bin": {
    "luis": "./luis.js"
  },
  "homepage": "https://github.com/tomitrescak/luis#readme",
  "dependencies": {
    "@tomino/toolbelt": "^1.1.3",
    "diff-view": "^1.0.12",
    "glob": "^7.1.3",
    "jest-diff": "^24.5.0",
    "jest-snapshots": "^1.1.2",
    "mobx": "*",
    "mobx-react": "*",
    "react-highlight": "^0.12.0",
    "react-split-pane": "^0.1.85",
    "semantic-ui-react": "*",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/jasmine": "^3.3.10",
    "@types/node": "^11.11.3",
    "@types/react": "^16.8.8",
    "@types/react-router": "^4.4.5",
    "fuse-box": "^3.7.1",
    "react": "*",
    "react-dom": "*",
    "tslint": "^5.14.0",
    "typescript": "*"
  }
}
