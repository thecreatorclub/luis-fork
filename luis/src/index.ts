export { renderLuis, Luis } from './client/components/index';
export { StateModel, mapBridgeToState } from './client/models/state_model';
export { Test } from './client/models/test_model';
export { TestGroup } from './client/models/test_group_model';
export { Snapshot } from './client/models/snapshot_model';
export { ProxyStore } from './client/models/proxy_store';
export { buildBridge } from './bridge';

import './types';
