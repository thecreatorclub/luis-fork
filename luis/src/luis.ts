#!/usr/bin/env node
// check if custom config exists
const fs = require('fs');
const path = require('path');

// parse parameters
let init = false;
let tests = '';
let root = 'src';
let entryFile = 'luis.';
let language = 'ts';
let customSet = false;
let start = true;
let bundler = 'fusebox';

function rtl(source: string) {
  const templatePath = require.resolve(`./templates/${source}`);
  // console.log('Loading: ' + templatePath);
  return fs.readFileSync(templatePath, { encoding: 'utf-8' });
}

function launch(program: string) {
  require('child_process').spawnSync(program, {
    stdio: 'inherit',
    shell: true,
    cwd: path.resolve('.')
  });
}

for (let i = 0; i < process.argv.length; i++) {
  let arg = process.argv[i];
  if (arg === 'init') {
    init = true;
  }
  if (arg === 'start') {
    start = true;
  }
  if (arg === '--tests') {
    tests = process.argv[i + 1];
    i++;
  }
  if (arg === '--language') {
    language = process.argv[i + 1];
    i++;
  }
  if (arg === '--bundler') {
    bundler = process.argv[i + 1];
    i++;
  }
  if (arg === '--root') {
    root = process.argv[i + 1];
    customSet = true;
    i++;
  }
  if (arg === '--run') {
    customSet = true;
    entryFile = process.argv[i + 1];
    i++;
  }
}

if (init) {
  console.log('‚úèÔ∏è  Initialising Luis');
  let file = path.resolve(path.join(root, entryFile + language));
  let dir = path.dirname(file);

  if (!fs.existsSync(dir)) {
    console.log('Root directory does not exist');
    console.log('üìÅ  Creating: ' + dir);
    fs.mkdirSync(dir, { recursive: true });
  }

  // load test file
  const configFile = tests ? rtl('entry_tests.js') : rtl('entry_no_tests.js');

  let configPath = path.resolve(file);
  if (fs.existsSync(configPath)) {
    console.log('üîß  Luis config already exists');
  } else {
    console.log('‚úÖ  Luis config created at: ' + file);
    fs.writeFileSync(configPath, configFile, { encoding: 'utf-8' });
  }

  if (tests === 'jest') {
    let testFilePath = path.resolve(`jest.config.js`);
    if (fs.existsSync(testFilePath)) {
      console.log('üîß  Jest config already exists');
    } else {
      let testFile = language === 'js' ? rtl('jest.config.js') : rtl('jest.config.ts');
      testFile = testFile
        .replace(/\$root/g, root)
        .replace('$ignore', `<rootDir>/${root}/summary.${language}`);
      fs.writeFileSync(testFilePath, testFile, { encoding: 'utf-8' });
      console.log('‚úÖ  Jest config created at: ' + testFilePath);
    }
  } else if (tests === 'mocha') {
    console.log('‚ùå  Mocha currently not supported');
  } else if (tests === 'jasmine') {
    console.log('‚ùå  Jasmine currently not supported');
  } else if (tests) {
    console.error('‚ùå  Test not supported: ' + tests);
  }

  let bundlerPath, bundlerConfig;
  if (bundler === 'fusebox') {
    bundlerPath = path.resolve('luis.fuse.js');
    bundlerConfig = rtl('luis.fuse.js')
      .replace('$root', root)
      .replace(/\$entry/g, entryFile + language);

    let entryPath = path.resolve('luis.fuse.html');
    if (!fs.existsSync(entryPath)) {
      let entryContent = rtl('luis.fuse.html');
      fs.writeFileSync(entryPath, entryContent, { encoding: 'utf-8' });
      console.log('‚úÖ  Fuse-Box entry created at: ' + entryPath);
    }
  } else if (bundler === 'parcel') {
  } else if (bundler === 'webpack') {
  } else if (bundler === 'cra') {
  } else {
    throw new Error('‚ùå  Bundler not supported: ' + bundler);
  }
  if (!fs.existsSync(bundlerPath)) {
    fs.writeFileSync(bundlerPath, bundlerConfig, { encoding: 'utf-8' });
    console.log('‚úÖ  Bundler config file created at: ' + bundlerPath);
  }
}

if (!init) {
  const fuseConfig = path.resolve('luis.fuse.js');
  const parcelConfig = path.resolve('luis.index.html');
  const webpackConfig = path.resolve('luis.webpack.js');
  const cra = path.resolve('node_modules/react-scripts');

  if (fuseConfig) {
    console.log('üöÄ  Found Fuse-Box config. Fusing Luis in 3..2..1');
    launch('node luis.fuse.js');
  } else if (parcelConfig) {
  } else if (webpackConfig) {
  } else if (cra) {
  } else {
    console.log(
      '‚ùå  Luis could not find a runtime file for any supported bundler. Please run "luis init" and see the documentation for all cli options'
    );
  }
}
