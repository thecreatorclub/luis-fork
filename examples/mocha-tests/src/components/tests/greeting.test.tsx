import * as React from 'react';

import { create } from 'react-test-renderer';

import { Greeting } from '../greeting';

// WARNING: Typescript crap
// this can be removed (it was creating problems for us when using types from Jest and Chai side by side)
import { expect } from 'chai';
// you have to import this at least once in your tests to pass the compilation
// this will register the global 'addStory' method
// in javascript you do not have to care about this
import 'luis/types';
// END WARNING

describe('Greeting', () => {
  const Component = () => <Greeting name="Tomas" />;

  it('renders correctly', () => {
    expect(create(<Component />)).to.matchSnapshot();
  });

  it('fails', () => {
    expect('123').to.equal('456');
  });

  it('fails multiline', () => {
    expect({
      a: 1,
      b: 2,
      c: 3
    }).to.equal({
      a: 1,
      b: 3,
      d: 4,
      c: 3,
      e: 6
    });
  });

  it('fails message', () => {
    expect(123).to.be.greaterThan(456);
  });

  addStory({
    component: Component
  });
});
