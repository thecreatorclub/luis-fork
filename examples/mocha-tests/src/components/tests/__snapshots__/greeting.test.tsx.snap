exports["Greeting renders correctly 1"] = "\n<div>\n  Hello1 \n  Tomas\n</div>\n";

